subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -169,7 +169,10 @@ public void visitDependencies(TaskDependencyResolveContext context) {
             parameterPropertyWalker.visitProperties(parameterObject, ParameterValidationContext.NOOP, new PropertyVisitor.Adapter() {
                 @Override
                 public void visitInputFileProperty(String propertyName, boolean optional, boolean skipWhenEmpty, @Nullable Class<? extends FileNormalizer> fileNormalizer, PropertyValue value, InputFilePropertyType filePropertyType) {
-                    context.maybeAdd(value.call());
+                    Object actualValue = value.call();
+                    if (actualValue != null) {
+                        context.maybeAdd(actualValue);
+                    }
```

Comment:
Why is this change necessary?

Reply:
The added test exposed a bug when the file property is not set (i.e. `.call` returns null`). Then null is passed to `maybeAdd` and the build fails when trying to determine the dependencies of `null`.